Use the PPPoE password to determine the system version

lordfriky opened the pull request:
All these changes were made by CelesteBlue, I'm just doing the PR since Celeste doesn't have a GitHub account anymore.

lordfriky committed:
Use PAP to determine the system version

TheOfficialFloW commented on pppwn.py:
```diff
@@ -181,13 +181,117 @@ def __init__(self, offs, iface, stage1, stage2):
+                if pkt[PPP_PAP_Request].passwd_len != 0:
+                    pwd = pkt[PPP_PAP_Request].password.decode("ascii")
+                    print("PAP password:", pwd)
+                    if pwd in ('750', '751', '755')\
```
Rewrite to not repeat the same list of fws

lordfriky replied:
I don't think we really need to know from where the firmware was selected, so I removed the check altogether. I can rewrite the other way if you prefer it.

lordfriky committed:
Remove repetitive firmware checking

lordfriky committed:
Add username back in case a second arg is needed

In case offsets differ from retail/devkit, for example.

Al-Azif committed:
Add more stuff for multi FW support

- < 5.50 needs less patching for copyin/copyout/copyinstr
- < 6.50 did not need veri patches
- < 6.50 does not require changes for ud2 instruction

Al-Azif commented:
Opened a PR with checks for multi FW stuff for lower FWs on this branch.

lordfriky committed:
Changes made by Celeste

lordfriky committed:
Merge branch 'celeste' into celeste

lordfriky committed:
Merge pull request #1 from Al-Azif/celeste

Add more stuff for multi FW support

lordfriky commented:
> Opened a PR with checks for multi FW stuff for lower FWs on this branch.

Merged, thank you!

VocalFan commented on pppwn.py:
```diff
@@ -147,6 +147,30 @@ def handler(self, pkt):
+    if key == "sys_ver":
+        return pwd
+
+def getOffsets(sys_ver):
+    if sys_ver in ('750', '751', '755'):
+        return OffsetsFirmware_750_755()
+    elif sys_ver in ('800', '801', '803'):
+        return OffsetsFirmware_800_803()
+    elif sys_ver in ('850', '852'):
+        return OffsetsFirmware_850_852()
+    elif sys_ver == '900':
+        return OffsetsFirmware_900()
+    elif sys_ver in ('903', '904'):
+        return OffsetsFirmware_903_904()
+    elif sys_ver in ('950', '951', '960'):
+        return OffsetsFirmware_950_960()
+    elif sys_ver in ('1000', '1001'):
+        return OffsetsFirmware_1000_1001()
+    elif sys_ver in ('1050', '1070', '1071'):
+        return OffsetsFirmware_1050_1071()
+    elif sys_ver == '1100':
+        return OffsetsFirmware_1100()
+    return None
```
```suggestion
def getOffsets(sys_ver):
    match sys_ver:
        case '750' | '751' | '755':
            return OffsetsFirmware_750_755()
        case '800' | '801' | '803':
            return OffsetsFirmware_800_803()
        case '850' | '852':
            return OffsetsFirmware_850_852()
        case '900':
            return OffsetsFirmware_900()
        case '903' | '904':
            return OffsetsFirmware_903_904()
        case '950' | '951' | '960':
            return OffsetsFirmware_950_960()
        case '1000' | '1001':
            return OffsetsFirmware_1000_1001()
        case '1050' | '1070' | '1071':
            return OffsetsFirmware_1050_1071()
        case '1100':
            return OffsetsFirmware_1100()
        case _:
            return None
```

Perhaps use a switch statement instead. Improves readability and also gives a slight perf improvement.

NelsonFrancisco replied:
Maybe use a map string -> function pointer

TheOfficialFloW commented on offsets.py:
```diff
@@ -205,6 +215,11 @@ class OffsetsFirmware_850_852:
 
     MEMCPY = 0xffffffff825a40f0
 
+    # Not available on FW >= 6.50
```
Why do you have these comments

Al-Azif replied:
Pre 6.50 you don't have the ud2 in that gadget, we can actually remove the `MOV_CR0_RSI_MOV_EAX_1_RET` gadget from FW >=6.50 and the `MOV_CR0_RSI_UD2_MOV_EAX_1_RET` from FW < 6.50. I just had them there when I was working on stuff so I could keep stuff straight, they're just the base @ ffffffff82200000 as a placeholder.

Or you can remove the conditional when the gadget is used and just use one of them for both (variable name won't match contents).

TheOfficialFloW replied:
Well please remove them

TheOfficialFloW commented on README.md:
```diff
@@ -65,18 +65,20 @@ On your PS4:
 - Choose `Custom` setup and choose `PPPoE` for `IP Address Settings`
-- Enter anything for `PPPoE User ID` and `PPPoE Password`
+- Enter anything for `PPPoE User ID`
+- Enter PS4/PS5 System Software version (e.g 900 for 9.00) for `PPPoE Password`
```
Why PS5??

TheOfficialFloW commented on pppwn.py:
```diff
@@ -147,6 +147,30 @@ def handler(self, pkt):
+    if key == "sys_ver":
+        return pwd
+
+def getOffsets(sys_ver):
```
get_offsets

Andryshik345 commented:
Will that fallback if i continue to use `--fw` parameter and random pppoe username-password?

tute123456 commented:
Hello, a particular version of Linux to use in the VM?
